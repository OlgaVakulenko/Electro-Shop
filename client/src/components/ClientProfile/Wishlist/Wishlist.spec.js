import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';

import { Wishlist } from './Wishlist';

const mockStore = configureStore([]);

describe('The test of Wishlist component', () => {
  let store;

  beforeAll(() => {
    store = mockStore({
      user: {
        name: 'Chuck',
      },
      wishlist: [1, 2, 3],
      wishlistProducts: [{
        name: 'addmotor hithot h1 sport mountain e-bike',
        itemNo: '2',
        id: '5db6fc84bfb42a414c724e7c',
        itemImg: '/img/products/e-bikes/2/001.jpg',
        price: 1699.99,
        url: '/products/2',
        rating: 4.6,
      }],
    });
  });

  it('Wishlist was rendered', () => {
    const rendered = shallow(
      <Provider store={store}>
        <Wishlist />
      </Provider>,
    );

    expect(rendered).toMatchSnapshot();
  });
});
