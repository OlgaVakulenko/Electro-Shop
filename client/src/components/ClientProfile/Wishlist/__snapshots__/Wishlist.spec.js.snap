// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The test of Wishlist component Wishlist was rendered 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Wishlist) />
</ContextProvider>
`;
